pkg/dio/writer.go

```diff
@@ -49,34 +50,41 @@ func (t *multiWriter) Write(p []byte) (int, error) {
 	}
 
 	t.mutex.Lock()
+
+	t.waitGroup.Add(1)
```

Comment:
If I am reading the other changes right, Close() will always be called after the last Write(). This may still be a good idea, but not sure if this contributes to the actual fix.